packages/api/errors.go

```diff
@@ -76,5 +82,47 @@ func NewAPIErrorWithResponse(operation string, res *resty.Response, additionalCo
 		apiError.ErrorMessage = errorMessage
 	}
 
+	if details != nil {
+		apiError.Details = details
+	}
+
 	return apiError
 }
+
+type errorResponse struct {
+	Message string   `json:"message"`
+	Details []string `json:"details"`
```

Comment:
details isn't necessarily array of string though right? 
![Uploading image.png…]()